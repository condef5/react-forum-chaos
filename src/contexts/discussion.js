import { createContext } from "react";

const DiscussionContext = createContext();

export const Provider = DiscussionContext.Provider;
export const Consumer = DiscussionContext.Consumer;
export default DiscussionContext;
