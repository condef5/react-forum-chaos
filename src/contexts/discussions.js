import React from "react";

const DiscussionContext = React.createContext();

export function DiscussionProvider({ discussions, setDiscussions, children }) {
  return (
    <DiscussionContext.Provider value={{ discussions, setDiscussions }}>
      {children}
    </DiscussionContext.Provider>
  );
}

export function useDiscussion() {
  return React.useContext(DiscussionContext);
}
