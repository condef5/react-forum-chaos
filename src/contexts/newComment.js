import { createContext } from "react";

const NewCommentContext = createContext();

export const Provider = NewCommentContext.Provider;
export const Consumer = NewCommentContext.Consumer;
export default NewCommentContext;
