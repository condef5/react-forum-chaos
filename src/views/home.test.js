import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Home from "./home";
import { DiscussionProvider } from "../contexts/discussions";

test("Home component", async () => {
  const discussions = [
    {
      id: new Date("2019-06-13T21:50:18.315Z").getTime(),
      title: "How kill to Christopher",
      body: "Some body in the sky",
      author: "condef5",
      date: "2019-06-13T21:50:18.315Z"
    },
    {
      id: new Date("2019-06-13T22:50:18.315Z").getTime(),
      title: "How to build the future today",
      body: "Some body in the sky",
      author: "condef5",
      date: "2019-06-13T22:50:18.315Z"
    }
  ];

  const setDiscussions = jest.fn();
  const { getByLabelText, asFragment } = render(
    <DiscussionProvider
      discussions={discussions}
      setDiscussions={setDiscussions}
    >
      <Home />
    </DiscussionProvider>
  );

  expect(asFragment()).toMatchSnapshot();

  const button = getByLabelText("Add new discussion");

  fireEvent.click(button);

  expect(asFragment()).toMatchSnapshot();

  fireEvent.click(button);

  fireEvent.keyDown(document, {
    key: "Escape",
    code: 27,
    charCode: 27
  });

  expect(asFragment()).toMatchSnapshot();
});
