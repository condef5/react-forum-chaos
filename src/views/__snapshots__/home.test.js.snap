// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component 1`] = `
<DocumentFragment>
  .emotion-4 {
  list-style: none;
  padding: 0;
  margin: 2em 0;
}

.emotion-1 {
  background-color: #fff;
  border-bottom: 1px solid #eee;
  padding: 20px 15px;
  line-height: 20px;
}

.emotion-1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 {
  color: #999;
  font-size: .85em;
}

.emotion-5 {
  position: fixed;
  bottom: 25px;
  right: 25px;
  border: none;
  border-radius: 50%;
  background-color: #f60;
  color: #fff;
  height: 55px;
  width: 55px;
  font-size: 35px;
}

<div>
    <ul
      class="emotion-4"
    >
      <li
        class="emotion-1"
      >
        <a
          href="/discussion/1560466218315"
        >
          How to build the future today
        </a>
        <div
          class="emotion-0"
        >
          <span>
            by condef5 
          </span>
          <span>
            6/13/2019
          </span>
          <span>
            | 1 comments
          </span>
        </div>
      </li>
      <li
        class="emotion-1"
      >
        <a
          href="/discussion/1560462618315"
        >
          How kill to Christopher
        </a>
        <div
          class="emotion-0"
        >
          <span>
            by condef5 
          </span>
          <span>
            6/13/2019
          </span>
          <span>
            | 1 comments
          </span>
        </div>
      </li>
    </ul>
    <button
      aria-label="Add new discussion"
      class="emotion-5"
    >
      +
    </button>
  </div>
</DocumentFragment>
`;

exports[`Home component 2`] = `
<DocumentFragment>
  .emotion-4 {
  list-style: none;
  padding: 0;
  margin: 2em 0;
}

.emotion-1 {
  background-color: #fff;
  border-bottom: 1px solid #eee;
  padding: 20px 15px;
  line-height: 20px;
}

.emotion-1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 {
  color: #999;
  font-size: .85em;
}

.emotion-5 {
  position: fixed;
  bottom: 25px;
  right: 25px;
  border: none;
  border-radius: 50%;
  background-color: #f60;
  color: #fff;
  height: 55px;
  width: 55px;
  font-size: 35px;
}

<div>
    <ul
      class="emotion-4"
    >
      <li
        class="emotion-1"
      >
        <a
          href="/discussion/1560466218315"
        >
          How to build the future today
        </a>
        <div
          class="emotion-0"
        >
          <span>
            by condef5 
          </span>
          <span>
            6/13/2019
          </span>
          <span>
            | 1 comments
          </span>
        </div>
      </li>
      <li
        class="emotion-1"
      >
        <a
          href="/discussion/1560462618315"
        >
          How kill to Christopher
        </a>
        <div
          class="emotion-0"
        >
          <span>
            by condef5 
          </span>
          <span>
            6/13/2019
          </span>
          <span>
            | 1 comments
          </span>
        </div>
      </li>
    </ul>
    <button
      aria-label="Add new discussion"
      class="emotion-5"
    >
      +
    </button>
  </div>
</DocumentFragment>
`;

exports[`Home component 3`] = `
<DocumentFragment>
  .emotion-4 {
  list-style: none;
  padding: 0;
  margin: 2em 0;
}

.emotion-1 {
  background-color: #fff;
  border-bottom: 1px solid #eee;
  padding: 20px 15px;
  line-height: 20px;
}

.emotion-1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 {
  color: #999;
  font-size: .85em;
}

.emotion-5 {
  position: fixed;
  bottom: 25px;
  right: 25px;
  border: none;
  border-radius: 50%;
  background-color: #f60;
  color: #fff;
  height: 55px;
  width: 55px;
  font-size: 35px;
}

<div>
    <ul
      class="emotion-4"
    >
      <li
        class="emotion-1"
      >
        <a
          href="/discussion/1560466218315"
        >
          How to build the future today
        </a>
        <div
          class="emotion-0"
        >
          <span>
            by condef5 
          </span>
          <span>
            6/13/2019
          </span>
          <span>
            | 1 comments
          </span>
        </div>
      </li>
      <li
        class="emotion-1"
      >
        <a
          href="/discussion/1560462618315"
        >
          How kill to Christopher
        </a>
        <div
          class="emotion-0"
        >
          <span>
            by condef5 
          </span>
          <span>
            6/13/2019
          </span>
          <span>
            | 1 comments
          </span>
        </div>
      </li>
    </ul>
    <button
      aria-label="Add new discussion"
      class="emotion-5"
    >
      +
    </button>
  </div>
</DocumentFragment>
`;
